lib/Receiver.js

```diff
@@ -590,37 +569,33 @@ var opcodes = {
       }
     },
     finish: function(mask, data) {
-      var self = this;
       var packet = this.unmask(mask, data, true) || new Buffer(0);
       var state = clone(this.state);
-      this.messageHandlers.push(function(callback) {
-        self.applyExtensions(packet, state.lastFragment, state.compressed, function(err, buffer) {
+      this.messageHandlers.push((callback) => {
+        this.applyExtensions(packet, state.lastFragment, state.compressed, (err, buffer) => {
           if (err) {
-            if (err.type === 1009){
-              return self.error('Max payload exceeded in compressed binary message. Aborting...', 1009);
-            }
-            return self.error(err.message, 1007);
+            this.error(err.message, err.closeCode === 1009 ? 1009 : 1007);
+            return;
           }
+
           if (buffer != null) {
-            var length = (self.currentMessageLength + buffer.length);
-            if (self.maxPayload == 0 || (self.maxPayload > 0 && length < self.maxPayload)) {
-              self.currentMessage.push(buffer);
-              self.currentMessageLength += length;
```

Comment:
There was a bug here as length was already the new length so it shouldn't have be summed but only assigned.